website/static/js/fangorn.js

```diff
@@ -38,6 +38,23 @@ $.extend(EXTENSION_MAP, {
     gsheet: 'xlsx'
 });
 
+var _defaultIconState = function () {
+    return {
+        mode : 'bar',
+        generalIcons : {
+            search : { on : true, template : searchIcon },
```

Comment:
Good point probably loose methods should be declared at the top with variables. Let me know if you have suggestions, code architecture is not my strongest area ... yet :)